mlflow/tracking/metic_value_conversion_utils.py

```diff
@@ -0,0 +1,163 @@
+import sys
+
+from mlflow.exceptions import MlflowException
+
+
+def is_module_imported(module_name: str) -> bool:
+    return module_name in sys.modules
+
+
+def __try_get_item(x):
+    try:
+        xval = x.item()
+    except ValueError:
+        raise MlflowException(
+            f"Expected metric value to contain a single element, got shape {x.shape} instead"
+        )
+
+    return xval
+
+
+def __converter_requires(module_name: str):
+    """Wrapper function that checks if specified `module_name` is already imported before invoking wrapped function."""
+
+    def decorator(func):
+        def wrapper(x):
+            if not is_module_imported(module_name):
+                return x
+
+            return func(x)
+
+        return wrapper
+
+    return decorator
+
+
+def convert_metric_value_to_str_if_possible(x) -> str:
+    if x is None or type(x) == str:
+        return x
+
+    converter_fns_to_try = [
+        convert_metric_value_to_str_if_pyspark_mlparam,
+        convert_metric_value_to_str_if_ndarray,
+        convert_metric_value_to_str_if_tensorflow_tensor,
+        convert_metric_value_to_str_if_torch_tensor,
+    ]
+
+    for converter_fn in converter_fns_to_try:
+        possible_str = converter_fn(x)
+        if type(possible_str) == str:
+            return possible_str
+
+    return str(x)
```

Comment:
Following from https://github.com/mlflow/mlflow/pull/5756/files#r866392045, I don't think we strictly need this method at this time. I also see that it's only called for parameter values (not metrics), so, at minimum, we should adjust the naming